core/src/main/java/org/jdbi/v3/core/extension/ExtensionHandler.java

```diff
@@ -110,11 +111,48 @@ static ExtensionHandler createForSpecialMethod(Method method) throws IllegalAcce
      * @return An {@link ExtensionHandler}
      */
     static ExtensionHandler createForMethodHandle(MethodHandle methodHandle) {
-        return (handleSupplier, target, args) -> {
-            if (target == null) {
-                throw new IllegalStateException("no target object present, called from a proxy factory?");
+        return new ExtensionHandler() {
+            @Override
+            public Invoker createInvoker(Object target) {
+                if (target == null) {
+                    throw new IllegalStateException("no target object present, called from a proxy factory?");
+                }
+                MethodHandle boundMh = methodHandle.bindTo(target);
+                return new Invoker() {
+                    @Override
+                    public Object invoke(HandleSupplier handleSupplier, Object... args) throws Exception {
+                        try {
+                            return boundMh.invokeWithArguments(args);
+                        } catch (Throwable e) {
+                            throw Sneaky.throwAnyway(e);
+                        }
+                    }
+                };
             }
-            return Unchecked.<Object[], Object>function(methodHandle.bindTo(target)::invokeWithArguments).apply(args);
         };
     }
+
+    /**
+     * Extension handler with a target object binding.
```

Comment:
I would not associate something called "Invoker" with "extension handler with a target object binding". I would have called that "BoundExtensionHandler" or something.